dspy/adapters/base.py

```diff
@@ -114,9 +113,11 @@ def _call_postprocess(
                 ]
                 value[tool_call_output_field_name] = ToolCalls.from_dict_list(tool_calls)
 
-            if citations and citation_output_field_name:
-                citations_obj = Citations.from_dict_list(citations)
-                value[citation_output_field_name] = citations_obj
+            # Parse custom types that does not rely on the adapter parsing
```

Comment:
maybe also moving the handling of `tool_calls` into the for loop?

Reply:
Yes, will do in a separate PR